content/blog/endpoint-deprecation/index.md

```diff
@@ -0,0 +1,115 @@
+---
+path: '/endpoint-deprecation'
+title: 'Can I deprecate this endpoint?'
+date: 2022-01-11T06:00:00.000+00:00
+featured_image: hero-image.png
+categories: ['Pixie Team Blogs', 'Kubernetes']
+authors: ['Hannah Troisi']
+emails: ['htroisi@pixielabs.ai']
+---
+
+Nothing lasts forever, including even the best designed APIs.
+
+Let’s imagine you are a developer who has taken over ownership of a Catalog microservice. You’ve been asked to deprecate the `/v1/catalog` endpoint in favor of the new `/v2/catalog` endpoint. How do you go about this?
+
+Whatever the reason for removal – a new version or a planned end-of-life – the first step in a _graceful_ API deprecation is to observe:
+
+- Is this endpoint used?
+- If so, who is calling it?
+
+## Is this endpoint used?
+
+Before you can deprecate the endpoint, you need to first check if the endpoint is actually being used.
+
+### Search the codebase
+
+For internal endpoints, a great way to start is to search the codebase for calls to the API. However, once you believe all calls have been removed, you will still want to use observability tooling to verify that all usage of the API has indeed stopped.
+
+Note that most established companies have standards for backwards compatibility of their microservice APIs (even internal ones). For example, a company might have a policy requiring 3 releases to pass between deprecation of an API and removal, in the event that there’s a rollback.
+
+### Verify with observability tooling
+
+Your company’s specific method for determining endpoint usage may vary. Some applications export metrics that they explicitly define on their services (e.g. Prometheus). Some applications are set up to log every inbound HTTP request (e.g. Apache logging).
+
+Another option is to use [Pixie](https://github.com/pixie-io/pixie), an open source observability tool for Kubernetes applications. Pixie automatically traces request traffic of [numerous protocols](https://docs.px.dev/about-pixie/data-sources/) (HTTP, MySQL, gRPC, and more) [using eBPF](https://docs.px.dev/about-pixie/pixie-ebpf/). But no matter how you gather the data, you’ll need to answer the same questions.
+
+Let’s check for HTTP traffic to the `/v1/catalog` endpoint.
```

Comment:
Maybe a smoother transition here is to tie the sentence back to the original question "Is this endpoint used"? 
Something like "Let's check for HTTP traffic to the `/v1/catalog` endpoint to see if there are any consumers of this endpoint."